feat: Date filters on bank reconciliation tool

sonali8848 opened the pull request:
https://user-images.githubusercontent.com/115159164/211467246-7cea9fc2-8a93-4af2-9c84-6f5f0eb35186.mp4

**Date filters in Bank Reconciliation Tool** 
Step 1-To apply date filter as per posting date in Bank Transactions and Vouchers and sorted bank transactions as per ascending order.
![Screenshot 2022-12-08 170131](https://user-images.githubusercontent.com/115159164/206436133-b13e59b9-98cf-4951-80f0-86f0c4b6b31b.png)
Step 2-  Added checkbox of 'filtered as per reference date' and two fields 'from reference date' and 'to reference date' to filter by reference date. If the checkbox is ticked then the vouchers will be filtered and sorted as per the reference date. 
![Screenshot 2022-12-08 172838](https://user-images.githubusercontent.com/115159164/206441284-855a18a7-d499-41f6-804d-112f931de2bc.png)
Step 3- Added field of posting date in the dialog.
![Screenshot 2022-12-08 173810](https://user-images.githubusercontent.com/115159164/208240544-713a643a-b83e-436d-834d-379ce13eeca7.png)
Step 4- Journal Entry checkbox is ticked by default.
![Screenshot 2022-12-08 173810](https://user-images.githubusercontent.com/115159164/208240580-2811b58c-6b7d-433a-96c7-44c1831cce2c.png)

**Consolidated auto bank reconciliation**

All the transactions will get auto-reconciled if the reference number matched with bank uploaded transaction and voucher posted, we can see that that pop message will show the count of transactions reconciled and if no transactions are reconciled then the pop will show “There is no matched reference number for reconciliation.
![image](https://user-images.githubusercontent.com/115159164/209940289-1472a060-6c01-4460-90fb-8c2d3a600668.png)
![image](https://user-images.githubusercontent.com/115159164/209940319-fb2f887f-cfc6-4bc4-802c-4825c42c8ffd.png)

sonali8848 committed:
Update bank_reconciliation_tool.py

Applying date filter on transactions and all the bank entries and also gives the filter the bank entries as per reference date. Sorted all transactions and entries as per date in ascending order.
Also added posting date columns in all bank entries and default checkbox tick of journal entry, hide the sales invoice and purchase invoice checkbox.

sonali8848 committed:
Filters on Bank Reconciliation

Applying date filter on transactions and all the bank entries and also gives the filter the bank entries as per reference date. Sorted all transactions and entries as per date in ascending order. Also added posting date columns in all bank entries and default checkbox tick of journal entry, hide the sales invoice and purchase invoice checkbox.

sonali8848 committed:
Update bank_reconciliation_tool.json

Adding fields in bank reconciliation tool

sonali8848 committed:
Feat:Filter on Payment Entries and Journal Entries

Applying filters on Payement entries and Journal Entries as per reference  date and posting date

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py:
```diff
@@ -25,6 +25,8 @@ class BankReconciliationTool(Document):
 @frappe.whitelist()
 def get_bank_transactions(bank_account, from_date=None, to_date=None):
 	# returns bank transactions for a bank account
+	from_date = frappe.db.get_single_value("Bank Reconciliation Tool", "bank_statement_from_date")
```
This is the incorrect way of getting the dates, pass from the client side. I can see options params are already there

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py:
```diff
@@ -51,6 +53,7 @@ def get_bank_transactions(bank_account, from_date=None, to_date=None):
 		],
 		filters=filters,
 	)
+	transactions = sorted(transactions, key=lambda x: x["date"]) if transactions else []
```
Can add a order by clause in query itself, this seems unnecessary

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py:
```diff
@@ -340,6 +343,9 @@ def get_linked_payments(bank_transaction_name, document_types=None):
 
 def check_matching(bank_account, company, transaction, document_types):
 	# combine all types of vouchers
+	filtered_by_reference_date = frappe.db.get_single_value(
```
Pass from client side, don't make a DB call

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py:
```diff
@@ -506,45 +520,103 @@ def get_lr_matching_query(bank_account, amount_condition, filters):
+	to_reference_date = frappe.db.get_single_value("Bank Reconciliation Tool", "to_reference_date")
+	filtered_by_reference_date = frappe.db.get_single_value(
+		"Bank Reconciliation Tool", "filtered_by_reference_date"
+	)
```
Pass from the client side

deepeshgarg007 commented on erpnext/hooks.py:
```diff
@@ -469,8 +469,8 @@
 	"Journal Entry",
-	"Purchase Invoice",
-	"Sales Invoice",
+	# "Purchase Invoice",
```
Why are they commented?

deepeshgarg007 commented on erpnext/public/js/bank_reconciliation_tool/dialog_manager.js:
```diff
@@ -35,13 +35,590 @@ erpnext.accounts.bank_reconciliation.DialogManager = class DialogManager {
+						reference_number: values.reference_number,
+						reference_date: values.reference_date,
+						party_type: values.party_type,
+						party: values.party,
+						posting_date: values.posting_date,
+						mode_of_payment: values.mode_of_payment,
+						project: values.project,
+						cost_center: values.cost_center,
+						allow_edit: true
+					},
+					callback: (r) => {
+						const doc = frappe.model.sync(r.message);
+						frappe.set_route("Form", doc[0].doctype, doc[0].name);
+					},
+				});
+			} else {
+				frappe.call({
+					method:
+						"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.create_journal_entry_bts",
+					args: {
+						bank_transaction_name: this.bank_transaction.name,
+						reference_number: values.reference_number,
+						reference_date: values.reference_date,
+						party_type: values.party_type,
+						party: values.party,
+						posting_date: values.posting_date,
+						mode_of_payment: values.mode_of_payment,
+						entry_type: values.journal_entry_type,
+						second_account: values.second_account,
+						allow_edit: true
+					},
+					callback: (r) => {
+						var doc = frappe.model.sync(r.message);
+						frappe.set_route("Form", doc[0].doctype, doc[0].name);
+					},
+				});
+			}
+		}
+	
+	};
```
What is this for, can't see any mention in the description. Please be more descriptive

deepeshgarg007 reviewed (changes requested):
- Assuming the user will be filtering either by reference or posting date. So when filter by reference date is enabled those fields should hide
- The commits are non-semantic, please add semantic comments
- This is the first level review, will add more once this cleanup is done

sonali8848 committed:
feat:filters on bank reconciliation

Added date filters on bank transactions, payment entries and journal entries and sorted list as per date in ascending order.

sonali8848 committed:
feat: added arguments of posting date and reference date

sonali8848 committed:
fix: linters

sonali8848 committed:
fix: json issue

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.js:
```diff
@@ -160,6 +160,9 @@ frappe.ui.form.on("Bank Reconciliation Tool", {
 					).$wrapper,
 					bank_statement_from_date: frm.doc.bank_statement_from_date,
 					bank_statement_to_date: frm.doc.bank_statement_to_date,
+					filtered_by_reference_date:frm.doc.filtered_by_reference_date,
+					from_reference_date:frm.doc.from_reference_date,
+					to_reference_date:frm.doc.to_reference_date,
```
```suggestion
					filtered_by_reference_date: frm.doc.filtered_by_reference_date,
					from_reference_date: frm.doc.from_reference_date,
					to_reference_date: frm.doc.to_reference_date,
```

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py:
```diff
@@ -504,47 +584,75 @@ def get_lr_matching_query(bank_account, amount_condition, filters):
 		currency_field = "paid_from_account_currency as currency"
+	filter_by_date = f"AND posting_date between '{from_date}' and '{to_date}'"
+	order_by = " posting_date"
+	if filtered_by_reference_date == "1":
```
Should be an int, why comparing with a string

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py:
```diff
@@ -504,47 +584,75 @@ def get_lr_matching_query(bank_account, amount_condition, filters):
 	cr_or_dr = "credit" if transaction.withdrawal > 0 else "debit"
+	# filter_by_date = f"AND je.posting_date between '{from_date}' and '{to_date}'"
+	order_by = " je.posting_date"
+	if filtered_by_reference_date == "1":
```
Here too

sonali8848 commented:
OKK , I will do

On Thu, Dec 29, 2022 at 11:14 AM Deepesh Garg ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
> ------------------------------
>
> In
> erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.js
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1058737905>:
>
> > +					filtered_by_reference_date:frm.doc.filtered_by_reference_date,
>
> +					from_reference_date:frm.doc.from_reference_date,
>
> +					to_reference_date:frm.doc.to_reference_date,
>
>
> ⬇️ Suggested change
>
> -					filtered_by_reference_date:frm.doc.filtered_by_reference_date,
>
> -					from_reference_date:frm.doc.from_reference_date,
>
> -					to_reference_date:frm.doc.to_reference_date,
>
> +					filtered_by_reference_date: frm.doc.filtered_by_reference_date,
>
> +					from_reference_date: frm.doc.from_reference_date,
>
> +					to_reference_date: frm.doc.to_reference_date,
>
>
> ------------------------------
>
> In
> erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1058738203>:
>
> >  	# get matching payment entries query
>
>  	if transaction.deposit > 0:
>
>  		currency_field = "paid_to_account_currency as currency"
>
>  	else:
>
>  		currency_field = "paid_from_account_currency as currency"
>
> +	filter_by_date = f"AND posting_date between '{from_date}' and '{to_date}'"
>
> +	order_by = " posting_date"
>
> +	if filtered_by_reference_date == "1":
>
>
> Should be an int, why comparing with a string
> ------------------------------
>
> In
> erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1058738311>:
>
> >  	# We have mapping at the bank level
>
>  	# So one bank could have both types of bank accounts like asset and liability
>
>  	# So cr_or_dr should be judged only on basis of withdrawal and deposit and not account type
>
>  	cr_or_dr = "credit" if transaction.withdrawal > 0 else "debit"
>
> +	# filter_by_date = f"AND je.posting_date between '{from_date}' and '{to_date}'"
>
> +	order_by = " je.posting_date"
>
> +	if filtered_by_reference_date == "1":
>
>
> Here too
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/erpnext/pull/33271#pullrequestreview-1232193625>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3OTA7GMROIDXIQ7HKF42YTWPUQLFANCNFSM6AAAAAASYB6TOE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

deepeshgarg007 commented:
@sonali8848 Please also add in the description why these changes were needed, what changes are added is well understood but why they were needed is still a bit confusing to me.

Unit tests also failing after the changes

sonali8848 commented:
Okk, I will add the proper description, but unit testing is failing because
we have passed the filters in get_linked_payments function, in  test cases
the filters are not passed that's why it is failing.


On Thu, Dec 29, 2022 at 11:26 AM Deepesh Garg ***@***.***>
wrote:

> @sonali8848 <https://github.com/sonali8848> Please also add in the
> description why these changes were needed, what changes are added is well
> understood but why they were needed is still a bit confusing to me.
>
> Unit tests also failing after the changes
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/erpnext/pull/33271#issuecomment-1367088746>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3OTA7HHXYM265E5QPIC463WPUR2BANCNFSM6AAAAAASYB6TOE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sonali8848 committed:
fix: filtered as per reference date

On bank reconciliation, transactions will be filtered as per date selected in 'from_date' and 'to_date' fields , In dialog, all the bank entries will  be fetched as per the posting date selected and if filtered by reference date checkbox is tick then then there will be two fields 'from_reference_date' and 'to_reference_date' then all bank entries in dialog box came as per reference date, selected. And by default journal entry checkbox is tick.
Also sorted the bank transactions and bank entries as per ascending order date wise.

deepeshgarg007 commented:
> Okk, I will add the proper description, but unit testing is failing because we have passed the filters in get_linked_payments function, in test cases the filters are not passed that's why it is failing

So pass the filters from the backend, is there any issue you are facing with this?

sonali8848 commented:
I will add the filters on test cases in test_bank_transaction

On Thu, Dec 29, 2022 at 12:10 PM Deepesh Garg ***@***.***>
wrote:

> Okk, I will add the proper description, but unit testing is failing
> because we have passed the filters in get_linked_payments function, in test
> cases the filters are not passed that's why it is failing
>
> So pass the filters from the backend, is there any issue you are facing
> with this?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/erpnext/pull/33271#issuecomment-1367103147>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3OTA7AGNHMPLJDD2MATUGLWPUW4JANCNFSM6AAAAAASYB6TOE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sonali8848 committed:
fix: pre-commit

sonali8848 committed:
fix: passing from_date and to_date filters in test cases

passing from_date and to_date filters in  test_linked_payments and test_debit_credit_output  for unit testing

sonali8848 committed:
fix: pre-commit

sonali8848 committed:
fix: pre-commit

sonali8848 committed:
feat: consolidated auto bank reconciliation

Added a button of Auto Reconcile, to reconcile the bank entries as per the matching reference number with the bank transaction and count of transactions reconciled message will be pop up on clicking the auto reconcile button.

sonali8848 committed:
fix: data format

sonali8848 commented:
Hi Deepesh,

The failed test cases are not linked with bank reconciliation, which we
have done. The link is mentioned below. Kindly check.
https://github.com/frappe/erpnext/actions/runs/3805772367/jobs/6498112341


On Fri, Dec 16, 2022 at 12:30 PM Deepesh Garg ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
>    - Assuming the user will be filtering either by reference or posting
>    date. So when filter by reference date is enabled those fields should hide
>    - The commits are non-semantic, please add semantic comments
>    - This is the first level review, will add more once this cleanup is
>    done
>
> ------------------------------
>
> In
> erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1050396085>:
>
> > @@ -25,6 +25,8 @@ class BankReconciliationTool(Document):
>  @frappe.whitelist()
>  def get_bank_transactions(bank_account, from_date=None, to_date=None):
>  	# returns bank transactions for a bank account
> +	from_date = frappe.db.get_single_value("Bank Reconciliation Tool", "bank_statement_from_date")
>
> This is the incorrect way of getting the dates, pass from the client side.
> I can see options params are already there
> ------------------------------
>
> In
> erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1050396503>:
>
> > @@ -51,6 +53,7 @@ def get_bank_transactions(bank_account, from_date=None, to_date=None):
>  		],
>  		filters=filters,
>  	)
> +	transactions = sorted(transactions, key=lambda x: x["date"]) if transactions else []
>
> Can add a order by clause in query itself, this seems unnecessary
> ------------------------------
>
> In
> erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1050396751>:
>
> > @@ -340,6 +343,9 @@ def get_linked_payments(bank_transaction_name, document_types=None):
>
>  def check_matching(bank_account, company, transaction, document_types):
>  	# combine all types of vouchers
> +	filtered_by_reference_date = frappe.db.get_single_value(
>
> Pass from client side, don't make a DB call
> ------------------------------
>
> In
> erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1050397029>:
>
> > +	from_date = frappe.db.get_single_value("Bank Reconciliation Tool", "bank_statement_from_date")
> +	to_date = frappe.db.get_single_value("Bank Reconciliation Tool", "bank_statement_to_date")
> +	from_reference_date = frappe.db.get_single_value(
> +		"Bank Reconciliation Tool", "from_reference_date"
> +	)
> +	to_reference_date = frappe.db.get_single_value("Bank Reconciliation Tool", "to_reference_date")
> +	filtered_by_reference_date = frappe.db.get_single_value(
> +		"Bank Reconciliation Tool", "filtered_by_reference_date"
> +	)
>
> Pass from the client side
> ------------------------------
>
> In erpnext/hooks.py
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1050399504>:
>
> > @@ -469,8 +469,8 @@
>  bank_reconciliation_doctypes = [
>  	"Payment Entry",
>  	"Journal Entry",
> -	"Purchase Invoice",
> -	"Sales Invoice",
> +	# "Purchase Invoice",
>
> Why are they commented?
> ------------------------------
>
> In erpnext/public/js/bank_reconciliation_tool/dialog_manager.js
> <https://github.com/frappe/erpnext/pull/33271#discussion_r1050400304>:
>
> > +				callback: (response) => {
> +					const alert_string = __("Bank Transaction {0} added as Payment Entry", [this.bank_transaction.name]);
> +					frappe.show_alert(alert_string);
> +					this.update_dt_cards(response.message);
> +					this.dialog.hide();
> +				},
> +			});
> +		}
> +	
> +		add_journal_entry(values) {
> +			frappe.call({
> +				method:
> +					"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.create_journal_entry_bts",
> +				args: {
> +					bank_transaction_name: this.bank_transaction.name,
> +					reference_number: values.reference_number,
> +					reference_date: values.reference_date,
> +					party_type: values.party_type,
> +					party: values.party,
> +					posting_date: values.posting_date,
> +					mode_of_payment: values.mode_of_payment,
> +					entry_type: values.journal_entry_type,
> +					second_account: values.second_account,
> +				},
> +				callback: (response) => {
> +					const alert_string = __("Bank Transaction {0} added as Journal Entry", [this.bank_transaction.name]);
> +					frappe.show_alert(alert_string);
> +					this.update_dt_cards(response.message);
> +					this.dialog.hide();
> +				},
> +			});
> +		}
> +	
> +		update_transaction(values) {
> +			frappe.call({
> +				method:
> +					"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.update_bank_transaction",
> +				args: {
> +					bank_transaction_name: this.bank_transaction.name,
> +					reference_number: values.reference_number,
> +					party_type: values.party_type,
> +					party: values.party,
> +				},
> +				callback: (response) => {
> +					const alert_string = __("Bank Transaction {0} updated", [this.bank_transaction.name]);
> +					frappe.show_alert(alert_string);
> +					this.update_dt_cards(response.message);
> +					this.dialog.hide();
> +				},
> +			});
> +		}
> +	
> +		edit_in_full_page() {
> +			const values = this.dialog.get_values(true);
> +			if (values.document_type == "Payment Entry") {
> +				frappe.call({
> +					method:
> +						"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.create_payment_entry_bts",
> +					args: {
> +						bank_transaction_name: this.bank_transaction.name,
> +						reference_number: values.reference_number,
> +						reference_date: values.reference_date,
> +						party_type: values.party_type,
> +						party: values.party,
> +						posting_date: values.posting_date,
> +						mode_of_payment: values.mode_of_payment,
> +						project: values.project,
> +						cost_center: values.cost_center,
> +						allow_edit: true
> +					},
> +					callback: (r) => {
> +						const doc = frappe.model.sync(r.message);
> +						frappe.set_route("Form", doc[0].doctype, doc[0].name);
> +					},
> +				});
> +			} else {
> +				frappe.call({
> +					method:
> +						"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.create_journal_entry_bts",
> +					args: {
> +						bank_transaction_name: this.bank_transaction.name,
> +						reference_number: values.reference_number,
> +						reference_date: values.reference_date,
> +						party_type: values.party_type,
> +						party: values.party,
> +						posting_date: values.posting_date,
> +						mode_of_payment: values.mode_of_payment,
> +						entry_type: values.journal_entry_type,
> +						second_account: values.second_account,
> +						allow_edit: true
> +					},
> +					callback: (r) => {
> +						var doc = frappe.model.sync(r.message);
> +						frappe.set_route("Form", doc[0].doctype, doc[0].name);
> +					},
> +				});
> +			}
> +		}
> +	
> +	};
>
> What is this for, can't see any mention in the description. Please be more
> descriptive
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/erpnext/pull/33271#pullrequestreview-1220317359>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3OTA7AFB26DEKYI7PKOXZLWNQHQPANCNFSM6AAAAAASYB6TOE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

deepeshgarg007 commented:
> @sonali8848 Please also add in the description why these changes were needed, what changes are added is well understood but why they were needed is still a bit confusing to me.

@sonali8848 this is still unanswered, can you update this as well

sonali8848 commented:
Hi Deepesh,

I have already added it in the comment section. The link is mentioned
below. Kindly check.
https://github.com/frappe/erpnext/pull/33271#issue-1484503677

Thanks


On Mon, Jan 2, 2023 at 4:08 PM Deepesh Garg ***@***.***>
wrote:

> @sonali8848 <https://github.com/sonali8848> Please also add in the
> description why these changes were needed, what changes are added is well
> understood but why they were needed is still a bit confusing to me.
>
> @sonali8848 <https://github.com/sonali8848> this is still unanswered, can
> you update this as well
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/frappe/erpnext/pull/33271#issuecomment-1368825273>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3OTA7AOOY256SEGCPCMMVLWQKVZPANCNFSM6AAAAAASYB6TOE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.js:
```diff
@@ -25,6 +25,24 @@ frappe.ui.form.on("Bank Reconciliation Tool", {
 		frappe.require("bank-reconciliation-tool.bundle.js", () =>
 			frm.trigger("make_reconciliation_tool")
 		);
+		frm.add_custom_button(__('Auto Reconcile'), function(){
```
```suggestion
		frm.add_custom_button(__('Auto Reconcile'), function() {
```

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.js:
```diff
@@ -25,6 +25,24 @@ frappe.ui.form.on("Bank Reconciliation Tool", {
 		);
+		frm.add_custom_button(__('Auto Reconcile'), function(){
+			frappe.call({
+				method:"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.auto_reconcile_vouchers",
```
```suggestion
				method: "erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.auto_reconcile_vouchers",
```

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.js:
```diff
@@ -25,6 +25,24 @@ frappe.ui.form.on("Bank Reconciliation Tool", {
+			frappe.call({
+				method:"erpnext.accounts.doctype.bank_reconciliation_tool.bank_reconciliation_tool.auto_reconcile_vouchers",
+				args: {
+					bank_account:frm.doc.bank_account,
```
```suggestion
					bank_account: frm.doc.bank_account,
```

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.js:
```diff
@@ -25,6 +25,24 @@ frappe.ui.form.on("Bank Reconciliation Tool", {
+					to_date: frm.doc.bank_statement_to_date,
+					filtered_by_reference_date:frm.doc.filtered_by_reference_date,
+					from_reference_date:frm.doc.from_reference_date,
+					to_reference_date:frm.doc.to_reference_date,
```
```suggestion
					filtered_by_reference_date: frm.doc.filtered_by_reference_date,
					from_reference_date: frm.doc.from_reference_date,
					to_reference_date: frm.doc.to_reference_date,
```

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py:
```diff
@@ -265,6 +266,82 @@ def create_payment_entry_bts(
+				}
+			)
+		# vouchers = frappe.as_json(voucherss)
+		# vouchers = json.loads(voucherss)
```
Remove comments

deepeshgarg007 commented on erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py:
```diff
@@ -265,6 +266,82 @@ def create_payment_entry_bts(
+		transaction.update_allocations()
+	matched_transaction_len = len(set(matched_transaction))
+	if matched_transaction_len == 0:
+		frappe.msgprint(_("There is no any matched reference number for reconciliation"))
```
```suggestion
		frappe.msgprint(_("No matching references found for auto reconciliation"))
```

deepeshgarg007 commented on erpnext/public/js/bank_reconciliation_tool/data_table_manager.js:
```diff
@@ -17,6 +22,8 @@ erpnext.accounts.bank_reconciliation.DataTableManager = class DataTableManager {
 			args: {
 				bank_account: this.bank_account,
+				from_date:this.bank_statement_from_date,
+				to_date:this.bank_statement_to_date
```
```suggestion
				from_date: this.bank_statement_from_date,
				to_date: this.bank_statement_to_date
```

deepeshgarg007 reviewed (changes requested):
Just some code cleanup changes, please add a simple test case for the Auto Reconciliation use case and also a working GIF in the description

sonali8848 committed:
fix: remove comments

deepeshgarg007 committed:
chore: fix fieldnames and order
